### ISSUE TITLE
Searches for anaconda3 packages in the wrong location

### ISSUE THREAD

**Issue by bfelbo** [Microsoft/pyright#100]
**Describe the bug**

I get the following lines in my VS Code Problems tab:

```
Import 'boto3' could not be resolved
Import 'psycopg2' could not be resolved
```

I'm running pyright on a working codebase that's using conda to manage packages. The local package environment has been activated with `conda activate aura` (aura is the name of the codebase and the corresponding local package environment).

Based on the output from pyright, it's searching for the packages in `/usr/local/anaconda3/envs/aura/lib/python3.6/`, which is not their actual location. 
I've found all of the desired packages in `/usr/local/anaconda3/envs/aura/lib/python3.6/site-packages/`.

**Screenshots or Code**

Debugging output for one of the libraries (boto3).

```
Could not import 'boto3' in file '/Users/bfelbo/code/aura/src/aura.py'
  Looking for typeshed path
  Using typeshed fallback path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.7'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.7/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.7/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.7/boto3.py'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.6'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.6/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.6/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.6/boto3.py'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.5'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.5/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.5/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3.5/boto3.py'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/3/boto3.py'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/2and3'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/2and3/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/2and3/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/stdlib/2and3/boto3.py'
  Typeshed path not found
  Attempting to resolve using root path '/Users/bfelbo/code/aura'
  Could not find directory '/Users/bfelbo/code/aura/boto3'
  Did not find file '/Users/bfelbo/code/aura/boto3.pyi' or '/Users/bfelbo/code/aura/boto3.py'
  Looking for typeshed path
  Using typeshed fallback path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback'
  Typeshed path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.19/typeshed-fallback/third_party' is not a valid directory
  Attempting to resolve using root path '/usr/local/anaconda3/envs/aura/lib/python3.6'
  Could not find directory '/usr/local/anaconda3/envs/aura/lib/python3.6/boto3'
  Did not find file '/usr/local/anaconda3/envs/aura/lib/python3.6/boto3.pyi' or '/usr/local/anaconda3/envs/aura/lib/python3.6/boto3.py'
```

**VS Code extension or command-line**
VS Code Extension 1.0.19

**Comment by erictraut** [comment]
Unfortunately, the output above wasn't sufficient to diagnose this problem. I just published 1.0.20, which contains more/better diagnostics. Please let me know if that helps you diagnose it. If not, please paste the output here, and I'll take a look.

**Comment by bfelbo** [comment]
Okay, I can try that. Wouldn't the simplest choice be to just add another path `current_search_path + "site-packages/"` to the directories, where you search for the relevant packages?

That should solve both this issue and #97.

**Comment by bfelbo** [comment]
The output with the newest version seems to be almost identical:

```
Could not import 'boto3' in file '/Users/bfelbo/code/aura/src/aura.py'
  Looking for typeshed stdlib path
  Using typeshed fallback path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.7'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.7/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.7/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.7/boto3.py'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.6'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.6/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.6/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.6/boto3.py'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.5'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.5/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.5/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3.5/boto3.py'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/3/boto3.py'
  Attempting to resolve using root path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/2and3'
  Could not find directory '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/2and3/boto3'
  Did not find file '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/2and3/boto3.pyi' or '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/stdlib/2and3/boto3.py'
  Typeshed path not found
  Attempting to resolve using root path '/Users/bfelbo/code/aura'
  Could not find directory '/Users/bfelbo/code/aura/boto3'
  Did not find file '/Users/bfelbo/code/aura/boto3.pyi' or '/Users/bfelbo/code/aura/boto3.py'
  Looking for typeshed third_party path
  Using typeshed fallback path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback'
  Typeshed path '/Users/bfelbo/.vscode/extensions/ms-pyright.pyright-1.0.20/typeshed-fallback/third_party' is not a valid directory
  Attempting to resolve using root path '/usr/local/anaconda3/envs/aura/lib/python3.6'
  Could not find directory '/usr/local/anaconda3/envs/aura/lib/python3.6/boto3'
  Did not find file '/usr/local/anaconda3/envs/aura/lib/python3.6/boto3.pyi' or '/usr/local/anaconda3/envs/aura/lib/python3.6/boto3.py'
```

**Comment by erictraut** [comment]
Pyright attempts to determine the search paths in a number of ways. In your case, it looks like it's falling back on asking the currently-configured python interpreter for its list of search paths. It does this by launching the interpreter and executing "import sys; print(sys.path)" and parsing the results. The search paths would normally include a path to the site-packages directory. Maybe conda leaves out this path and the python interpreter adds it at runtime? Could you do me a favor and validate that theory by activating your python environment from a command line and then running `python -c "import sys; print(sys.path)`. Does the output include a path that ends in `site-packages`?

**Comment by bfelbo** [comment]
Here's `python -c "import sys; print(sys.path)"` before running `conda activate aura`:

```
['', '/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python27.zip', '/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7', '/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-darwin', '/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac', '/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac/lib-scriptpackages', '/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk', '/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-old', '/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload', '/Users/bfelbo/Library/Python/2.7/lib/python/site-packages', '/usr/local/lib/python2.7/site-packages']
```

And here's `python -c "import sys; print(sys.path)"` after running `conda activate aura`:

```
['', '/usr/local/anaconda3/envs/aura/lib/python36.zip', '/usr/local/anaconda3/envs/aura/lib/python3.6', '/usr/local/anaconda3/envs/aura/lib/python3.6/lib-dynload', '/usr/local/anaconda3/envs/aura/lib/python3.6/site-packages']
```

**Comment by bfelbo** [comment]
So yes, it does include the `site-packages` path

**Comment by erictraut** [comment]
OK, thanks. I found the bug. The code was failing to search beyond the first of multiple search paths returned by the python interpreter.

**Comment by erictraut** [comment]
This should now be fixed in version 1.0.21, which I just published. Thanks again for the bug report and the details that helped me diagnose the problem. Let me know if you run into any other problems with pyright.
